### PR TITLE
FES: _validateParameters() ver 1.0 for release

### PR DESCRIPTION
**Pull request by almchung**
#### `core/error_helpers/validateParameters()`:
* can be called through `p5._validateParameters(FUNCT_NAME, ARGUMENTS)` 
or, `this._validateParameters(FUNCT_NAME,  arguments)` within the function.
* p5.js objects will need to assign value for `name` parameter (name of the object) within the class declaration.

### inline-documentation updates
- mainly for `src/color/creating_reading.js` functions.

### wiki documentation
https://github.com/processing/p5.js/wiki/Friendly-Error-System
will be updated soon!

### new unit tests
- test.html: unit tests for 2d_primitives should be uncommented
- `test/unit/color/color_conversion.js`
- `test/unit/color/creating_reading.js`
- `test/unit/color/p5.Color.js`
- `test/unit/color/setting.js`
- `test/unit/core/2d_primitives.js` - removed pixel comparison tests
- `test/unit/core/curves.js`
- `test/unit/core/error_helpers.js`
- `test/unit/utilities/string_functions.js`

### CODE REVIEW

**Comment by lmccart** [comment]
hey @almchung could you pull the latest from master and resolve any conflicts? looks like it's mostly small things.

**Comment by almchung** [comment]
will do that starting this afternoon!

File: src/core/error_helpers.js [hunk 1]
```diff
@@ -53,23 +57,23 @@ function report(message, func, color) {
     color = typeColors[color];
   }
   // LM TEMP commenting this out until we get the whole system working
```
Comment (lmccart) [inline src/core/error_helpers.js]: this comment can be removed now :)

File: src/core/error_helpers.js [hunk 2]
```diff
@@ -253,6 +406,8 @@ function helpForMisusedAtTopLevelCode(e, log) {
 
 // Exposing this primarily for unit testing.
 p5.prototype._helpForMisusedAtTopLevelCode = helpForMisusedAtTopLevelCode;
+p5.prototype._validateParameters = validateParameters;
+p5._validateParameters = validateParameters;
```
Comment (lmccart) [inline src/core/error_helpers.js]: do we need both lines here?
and can we just put this in the definition above? ie
p5._validateParameters = function(......

File: src/core/error_helpers.js (hunk 1 above)
Comment (almchung) [inline src/core/error_helpers.js]: Yay, done!

File: src/core/error_helpers.js (hunk 2 above)
Comment (almchung) [inline src/core/error_helpers.js]: Done reformatting: `p5._validateParameters = function(...`
Linking the function under `prototype` was mainly for bypassing linking issues which occurs during unit tests. I will later double-check and update the unit tests if I can find a good solution.

**Comment by almchung** [comment]
@lmccart  just finished resolving conflicts with `master` branch!

File: src/core/error_helpers.js (hunk 2 above)
Comment (lmccart) [inline src/core/error_helpers.js]: got it. when you have a chance, can you add a comment here so we remember why it's like this

**Comment by lmccart** [comment]
hooray!! I will put out a new release in the next few days so we can start testing more widely.

**Pull request merged**